### ISSUE TITLE
Index out of range in CalendarCollectionViewSource.itemAt

### ISSUE THREAD

**Issue by arturs105** [toggl/mobileapp#7130]
## 💥 Crash Report
One of the few remaining crashes that we can fix. Amounts to almost half of all crashes in iOS 2.11.

### 👥 Impact
Has occured 38 times to 25 users

### 🔗 Context
Has happened only on iOS 13.2 and up.

### 🛰 Link
https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/errors/1047824208u/overview

### 📝  Exception & Stack Trace
`SIGABRT: Index was out of range. Must be non-negative and less than the size of the collection. Parameter name: index`
```cs
List`1[T].get_Item (System.Int32 index)
CalendarCollectionViewSource.itemAt (System.Int32 index)
CalendarCollectionViewSource.UpdateItemView (System.DateTimeOffset startTime, System.Nullable`1[T] duration)
CalendarCollectionViewSource.UpdateItemView (Toggl.Core.Calendar.CalendarItem calendarItem)
AnonymousSafeObserver`1[T].OnNext (T value)
(wrapper other) System.Object.gsharedvt_out()
Sink`1[TTarget].ForwardOnNext (TTarget value)
(wrapper other) System.Object.gsharedvt_in()
(wrapper other) System.Object.gsharedvt_out()
IdentitySink`1[T].OnNext (T value)
(wrapper other) System.Object.gsharedvt_in()
Subject`1[T].OnNext (T value)
CalendarCollectionViewEditItemHelper.changeEndTime (CoreGraphics.CGPoint point)
CalendarCollectionViewEditItemHelper.panChanged (CoreGraphics.CGPoint point)
CalendarCollectionViewEditItemHelper.onPan (UIKit.UIPanGestureRecognizer gesture)
UIPanGestureRecognizer+Callback.Activated (UIKit.UIPanGestureRecognizer sender)
(wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate)
UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName)
Application.Main (System.String[] args)
```

**Comment by juanlaube** [comment]
I think I accidentally fixed this in #6748 🤔

**Comment by juanlaube** [comment]
Yep, this was solved in another PR 😄
